server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -201,12 +202,12 @@ public Boolean valueForDisplay(Object value) {
 
         @Override
         public DocValueFormat docValueFormat(@Nullable String format, DateTimeZone timeZone) {
+            this.name();
             if (format != null) {
-                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support custom formats");
+                this.failUnsupportedFeature("custom formats");
```

Comment:
Quite a lot of places had the same "copy/paste" of this line so i extracted it into the a base class.